benchmark/framework/containerd_utils.go

```diff
@@ -0,0 +1,304 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package framework
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+)
+
+var (
+	testNamespace               = "BENCHMARK_TESTING"
+	testContainerId             = "TEST_RUN_CONTAINER"
+	testEnvironment             = "TEST_RUNTIME"
+	outputFilePerm  fs.FileMode = 0777
```

Comment:
0666 would be more accurate as these are not executable

Reply:
Bit confused about who would work on that :)

Also I'm fine just hardcoding these numbers on os.Mkdir() and/or os.Create().